Sources/PIALibrary/Account/DefaultAccountProvider.swift

```diff
@@ -503,26 +514,32 @@ open class DefaultAccountProvider: AccountProvider, ConfigurationAccess, Databas
             self.accessedDatabase.secure.setUsername(credentials.username)
             self.accessedDatabase.secure.setPassword(credentials.password, for: credentials.username)
 
-            self.webServices.token(credentials: credentials) { (error) in
-                if error != nil {
-                    callback?(nil, error)
-                    return
-                }
-
-                self.webServices.info() { (accountInfo, error) in
-                    guard let accountInfo = accountInfo else {
-                        callback?(nil, error)
-                        return
-                    }
-
-                    self.accessedDatabase.plain.accountInfo = accountInfo
-                    self.accessedDatabase.secure.setPublicUsername(accountInfo.username)
-                    
-                    let user = UserAccount(credentials: credentials, info: nil)
-                    Macros.postNotification(.PIAAccountDidSignup, [.user: user])
-                    callback?(user, nil)
-                }
-            }
+            
+            // This method `loginWithCredentials` executes all the API calls and makes the same updates as the code commented out below.
+            // TODO: Make sure that this works as expected when finishing the integration of the sign up API
+            self.loginWithCredentials(credentials, notificationToSend: .PIAAccountDidSignup, callback: callback)
```

Comment:
@kp-said-rehouni , since the signup API also makes use of the login and accountDetails APIs that are replaced with the new implementation in this PR, I commented out the code from the old implementation and replaced it with this new method `loginWithCredentials` that performs both requests using the new UseCases

Let me know your thoughts. You will probably have to adjust or rebase this updates onto your branch.